### ISSUE TITLE
XP support and/or artifacts vs releases and/or mingw-64 compile

### ISSUE THREAD

**Issue by vogonsorg** [coelckers/prboom-plus#424]
Hello,
I've setup a fork looking into XP compatibility so if that's not something you have any concern for then feel free to close this. If you do then..... https://github.com/vogonsorg/prboom-plus/issues/2

It looks like after 2.5.17um XP support was dropped and it looks like the dependencies libzstd.dll, libwebp-7.dll and libwinpthread-1.dll are the culprit.

I noticed long ago that pacman packages are usually compiled with mingw-w64 w/ posix thread instead of win32 thread. A couple of years ago mingw-w64 broke XP compatibility that requires a reversal of a commit to maintain compatibility or a win32 thread compiled mingw-w64 (available by default in linux repos or configurable when installing MSYS2) can be used. The issue with using MSYS2 is that unless it's been fixed and I haven't been paying attention which is likely because I don't use MSYS2 then you can use a win32 thread mingw-w64 when installing MSYS2 but the pacman packages for dependencies are compiled with posix.

What's confusing is that it looks like in Actions that the artifacts are compiled with "Windows_Dependencies_v4.zip" (old versions of the dependencies that work ) but the actual releases are using newer versions of the dependencies download with vcpkg (new versions of dependencies that don't work) ?. So there is a disconnect between the artifacts and the releases that needs to be corrected.

It also looks like there isn't a MSYS2 or Linux mingw-w64 cross compile for prboom-plus so pr-boom plus is compiled with visual studio but the dependencies are compiled with mingw? Would be nice if both matched. Either pr-boom plus and dependecies compiled with Visual Studio or Mingw-w64.

For XP compatibility may be better to use mingw-w64 under linux and cross compile to Windows. Haven't bothered to do that with Github but do that all the time in WSL or Linux.

TLDR If not concerned about XP compatibility: May be worth fixing the artifacts and releases disconnect and adding mingw-w64 for compiling pr-boom plus

**Comment by fabiangreffrath** [comment]
The libraries that the CI builds link against are not necessarily the same ones used for the releases.

**Comment by RamonUnch** [comment]
I am not sure about the main prboom-plus-um, but I am making builds of [dsda-doom](https://github.com/RamonUnch/dsda-doom) that are compatible with Windows XP. I am not doing anything special, I just use MSYS1 (that I installed a long time ago on my xp computer) and I build from here.
I use:
MSYS 1
CMAKE 3.12.4.0
TDMGCC 9.2.0 (mingw32 based), but I guess it would work with mingw64.

I installed manually all the usefull lins in `/usr/libs` libs.
I was not expecting it to actually work but it does.
